src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double RESIZE_FACTOR = 1.5;
+
+    private T[] container;
+    private int size;
+
+    public ArrayList() {
+        container = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growArray(size + 1);
+        container[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateIndexForAdd(index);
+        growArray(size + 1);
+        insertElement(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return container[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validateIndex(index);
+        container[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        T removedElement = container[index];
+        removeElement(index);
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = -1;
+        for (int i = 0; i < size; i++) {
+            if (elementsAreEqual(element, container[i])) {
+                index = i;
+                break;
+            }
+        }
+        if (index == -1) {
+            throw new NoSuchElementException("Can't find element while removing");
+        }
+        return remove(index);
```

Comment:
rewrite this without using index local var, second if and break
in such  cases, cleaner code would be returning inside of loop, throwing outside of loop (without if, because if that code is getting executed, it means we didn't find the match)